### PR TITLE
Github Test Restructuring 

### PR DESCRIPTION
**Pull request by sidpremkumar**
Restructuring and recreating Github tests

### CODE REVIEW

File: test/github_tests/test_github.py [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+            uname=mock_uname,
+            repo_name="dummy_repo")
+
+        # Validate function calls and response
+        mock_uname.get_repo.assert_called_with('dummy_repo')
+        self.assertEqual([], response)
+
+    @patch(PATH + 'GithubService.check_request_state')
+    @patch(PATH + 'GithubService.get_last_comment')
+    @patch(PATH + 'GithubReview')
```
[the remark this answers is not in the document]
Comment (sidpremkumar) [reply test/github_tests/test_github.py]: I've added a check to assert that it was called with the correct arguments

File: test/github_tests/test_github.py [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+    def test_get_reviews_get_repo_fails(self):
+        """
+        Tests get_reviews function where getting the repo fails
+        """
+        # Set up mock return values and side effects
+        mock_uname = MagicMock()
+        mock_uname.get_repo.return_value = Exception()
+
+        # Call function
+        with self.assertRaises(Exception):
```
[the remark this answers is not in the document]
Comment (sidpremkumar) [reply test/github_tests/test_github.py]: I am trying to mock this exception here: 
`# get authenticated github object
        g = Github(token)
        log.debug('Github instance created: %s', g)
        try:
            # get user object
            uname = g.get_user(user_name)
        except UnknownObjectException:
            log.exception('Invalid username/organizaton: %s', user_name)
            raise Exception('Invalid username/organizaton: %s' % user_name)`

As the Exception raised is within the review-rot code and not the GibhubService code would this Exception be fine? Or would I need to create a custom exception?

[the remark this answers is not in the document]
Comment (Lukkra211) [reply test/github_tests/test_github.py]: Ah, yeah, didn't noticed that. In that case, it is fault of the code... Let's keep the tests as they are right now then.

File: test/github_tests/test_github.py [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+    from mock import patch  # noqa: F401
+try:
+    # Python 3.3 >
+    from unittest.mock import MagicMock  # noqa: F401
+except ImportError:
+    from mock import MagicMock  # noqa: F401
+
+
+from github.GithubException import UnknownObjectException
+
```
[the remark this answers is not in the document]
Comment (sidpremkumar) [reply test/github_tests/test_github.py]: I am following the import setup that Lukas suggested: 
The order of imports should usually be divided into three groups. First group are imports using build-in modules, second group should consists imports using 3rd party modules, and last one should contains project imports. 

I will removed the white space between them

**Comment by amarza-rh** [comment]
Also test this branch:

```python
            if last_issue_comment.created_at > last_review_comment.created_at: 
                last_comment = last_issue_comment
```

**Pull request merged**